pkgs/development/interpreters/python/cpython/default.nix

```diff
@@ -513,7 +514,24 @@ in with passthru; stdenv.mkDerivation {
 
   separateDebugInfo = true;
 
-  inherit passthru;
+  passthru = passthru // {
+    doc = stdenv.mkDerivation {
+      inherit src;
+      name = "python${pythonVersion}-${version}-doc";
+
+      dontConfigure = true;
+
+      dontBuild = true;
+
+      sphinxRoot = "Doc";
+
+      postInstallSphinx = ''
+        mv $out/share/doc/* $out/share/doc/python${pythonVersion}-${version}
+      '';
+
+      nativeBuildInputs = with pkgsBuildBuild.python3.pkgs; [ sphinxHook python_docs_theme ];
+    };
```

Comment:
Hmmm, I don't think I've seen other packages having a doc passthru attributes. Usually, the doc is placed in the `$doc` output instead. I think it would be less confusing to use a [multiple outputs derivation](https://nixos.org/manual/nixpkgs/stable/#sec-multiple-outputs-)

Reply:
Can't. It will have cause horrible recursion issues. See https://github.com/NixOS/nixpkgs/pull/229569#issuecomment-1557529639